### PR TITLE
Feature/add plans on calendar

### PR DESCRIPTION
**Pull request by pandafulmanda**
screen shot 2015-04-22 at 6 26 14 pm
screen shot 2015-04-22 at 6 26 17 pm

- [x] Add iReading and Add Homework as popup
- [x] Link to add functionality
- [x] tests
  - [x] clicking + shows add plan links
  - [x] clicking on an upcoming day shows add plan links
  - [x] clicking on past days not allowed
- [ ] get rid of annoying warnings -- TODO/new PR

### CODE REVIEW

**Comment by pandafulmanda** [comment]
calendar may not look a like a calendar, waiting for #111 to get merged

**Comment by philschatz** [comment]
Hmm, the link to add only shows up on some dates (23-30) for me. Here's what my calendar looks like:

image

(Also noticed the stagger but that might be a separate PR)

**Comment by pandafulmanda** [comment]
I purposely disallow adding on past dates, but we can easily change that

**Comment by philschatz** [comment]
oh, that makes sense. Maybe a question for @Fredasaurus (in another PR if it needs to be changed)

File: src/components/course-calendar/month.cjsx [hunk 1]
```diff
@@ -38,8 +45,8 @@ CourseMonth = React.createClass
     nodesWithHeights = calendar.querySelectorAll('.rc-Week')
 
     Array.prototype.forEach.call(nodesWithHeights, (node, nthRange) ->
-        range = _.findWhere(groupedDurations, {nthRange: nthRange})
-        node.style.height = range.dayHeight + 'rem'
+      range = _.findWhere(groupedDurations, {nthRange: nthRange})
+      node.style.height = range.dayHeight + 'rem'
```
Comment (philschatz) [inline src/components/course-calendar/month.cjsx]: Can you document in the LESS the node that will get this height? and include a comment in here with a greppable string to be able to find it easily? Something like `CALENDAR_DAY_DYNAMIC_HEIGHT`? There seem to be more of these special-style cases showing up in the code and I'm worried it'll be difficult to find them in 6mos.

File: src/components/course-calendar/month.cjsx [hunk 2]
```diff
@@ -52,21 +59,77 @@ CourseMonth = React.createClass
...
+    unless mouseEvent.relatedTarget is React.findDOMNode(@refs.addOnDay)
+      @hideAddOnDay(componentName, dayMoment, mouseEvent)
+
+  undoActives: (componentName, dayMoment, mouseEvent) ->
+    unless dayMoment? and dayMoment.isSame(@refs.addOnDay.state.date, 'day')
+      @hideAddOnDay(componentName, dayMoment, mouseEvent)
+
+  hideAddOnDay: (componentName, dayMoment, mouseEvent) ->
+    @refs.addOnDay.close()
+    document.querySelector('.active.rc-Day')?.classList.remove('active')
```
Comment (philschatz) [inline src/components/course-calendar/month.cjsx]: Any reason this isn't in the related React component's state?

File: src/components/course-calendar/month.cjsx (hunk 1 above)
Comment (philschatz) [inline src/components/course-calendar/month.cjsx]: Resolved; Thanks!

**Pull request merged**